### DISCUSSION TITLE
Search using SearXNG via the command line

### DISCUSSION THREAD

**Discussion by MichaelDoron** [searxng/searxng#2169] [Q&A]
Hi all, 

Is there a way to search the internet using SearXNG via the commandline, without opening a graphical browser?
Sorry if this was addressed before, I couldn't find a similar discussion in the Q&A, documentation or git issues -- all I find is the command engine, that seems to be for searching the file system.

Thanks!

**Comment by glanham-jr** [comment]
What is your use specific use case for command line? If its just preference, there are terminal browsers like W3M / Lynx you could use with variable mileage. Not sure if you would prefer an API available or just as a super quick "Give me the top X results, brief description, and the URL".

**Comment by hrizal** [comment]
This is a very good library, yes I also need it, from the quality search results (a combination of several search engines), then I will open the pages one by one and make the summary (for example with the sumy module), then again I feed it to gpt4all, and of course all in all I'm just combining the power of each library, to produce great shared services in other forms

**Comment by glanham-jr** [comment]
I see two steps to developing a CLI utility.

1. We create a `cli.py` file somewhere of this repo which simply provides CLI interface to some of code. This would let you search the internet, but it would just be on your local machine. Defeats the purpose of anonymity, but it may be good enough for your use cases.
2. Would have to double check, but we would need public API's which return JSON (or some other data). Then, the CLI tool could accept a searx url, and you could pass in the same parameters as to 1, but would call an HTTP endpoint. 

Option 1 could possibly be developed as an additional tool in the repo, but Option 2 is somewhat of a different project and doesn't belong in this specific repo. I would say Option 1 could be developed first as an experiment tool, and if there is continued interest in it, it could be developed more.
